download-basket/src/main/java/downloadbasket/helpers/OGCServices.java

```diff
@@ -0,0 +1,208 @@
+        JSONArray identifiers = new JSONArray(download.getString(PARAM_IDENTIFIERS));
+        StringWriter s = new StringWriter();
+        StringWriter filter = new StringWriter();
+        String croppingNameSpace = PropertyUtil.get("hsy.wfs.cropping.namespace");
+
+        try {
+            if(identifiers.length()>1) {
+                if(!addNameSpaces) {
+                    filter.append("<ogc:Or>");
+                } else {
+                    filter.append("<Or xmlns:ogc=\"http://www.opengis.net/ogc\">");
+                }
+            }
+
+            String intersectStart = "<ogc:Intersects>";
+            if(identifiers.length()==1 && addNameSpaces) {
+                intersectStart = "<Intersects xmlns:ogc=\"http://www.opengis.net/ogc\">";
+            }
+            String intersectEnd = "</ogc:Intersects>";
+            if(identifiers.length()==1 && addNameSpaces) {
+                intersectEnd = "</Intersects>";
+            }
+
+            for(int id=0;id<identifiers.length();id++){
+                JSONObject identifier = identifiers.getJSONObject(id);
+                String layerName = Helpers.getLayerNameWithoutNameSpace(identifier.getString("layerName"));
+                String uniqueColumn = identifier.getString("uniqueColumn");
+                String uniqueValue = identifier.getString("uniqueValue");
+                String cropGeomColumn = identifier.getString("geometryName");
+                String filterColumnType = identifier.getString("geometryColumn");
+                filter.append(intersectStart);
+                filter.append("<ogc:PropertyName>"+cropGeomColumn+"</ogc:PropertyName>");
+                filter.append("<ogc:Function name=\"querySingle\">");
+                filter.append("<ogc:Literal>"+croppingNameSpace+":"+layerName+"</ogc:Literal>");
+
+                filter.append("<ogc:Literal>"+cropGeomColumn+"</ogc:Literal>");
+                if(filterColumnType.equals("STRING")) {
+                    filter.append("<ogc:Literal>"+uniqueColumn + " LIKE '" + uniqueValue.trim() +"%" + "'</ogc:Literal>");
+                } else {
+                    filter.append("<ogc:Literal>"+uniqueColumn + " =  " + uniqueValue + "</ogc:Literal>");
+                }
+                filter.append("</ogc:Function>");
+                filter.append(intersectEnd);
+            }
+
+            if(identifiers.length()>1){
+                if(!addNameSpaces) {
+                    filter.append("</ogc:Or>");
+                } else {
+                    filter.append("</Or>");
+                }
+            }
+
+            if(writeParam){
+                s.append("&filter=");
+            }
+
+            s.append(URLEncoder.encode(filter.toString(), "UTF-8"));
+
+            LOGGER.debug("Created plugin filtter:" + filter.toString());
+
+        } catch(Exception e){
+            e.printStackTrace();
```

Comment:
Use logger instead. Also you can just return "" instead of creating a StringWriter to do it.